src/ParseType.elm

```diff
@@ -94,6 +98,14 @@ extractBasic encoding txt =
     map (detectExtendedRecord declared) declared
 
 
+
+{-| Don't know what this does
```

Comment:
Probably I can update this comment. What is the encoding parameter for?

Reply:
It signals whether you're scanning types for encoding (True) or for decoding (False). When you're encoding something like "type Record = { field : Maybe Int }",  you need a separate encoder for Maybe Int, because there's no Encode.maybe (but there's a Decode.maybe).

The "Either a b" result looks buggy...

Reply:
I am not sure if I understand. Why do you need that information when scanning the types? Shouldn't that only be relevant when generating? :)

Reply:
It would be more elegant to check for this at the generating stage, I agree. I'll look at it.